airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py

```diff
@@ -135,7 +135,15 @@ def build_stream_processor(
                 primary_key = get_field(configured_stream, "primary_key", f"Undefined primary key for stream {stream_name}")
 
             message = f"'json_schema'.'properties' are not defined for stream {stream_name}"
-            properties = get_field(get_field(stream_config, "json_schema", message), "properties", message)
+            schema = get_field(stream_config, "json_schema", f"'json_schema' is not defined for stream {stream_name}")
+            if "properties" in schema:
+                properties = get_field(schema, "properties", message)
+            elif "oneOf" in schema:
+                options = filter(lambda option: "properties" in option, schema["oneOf"])
+                if len(options) == 0:
+                    raise KeyError(f"Stream {stream_name} does not have any properties")
+                # If there are multiple oneOf options, just pick the first one - we don't really support oneOf to begin with
```

Comment:
Ehm... is there an issue that tracks supporting `oneOf` or disabling `oneOf` for sources? Aren't we always risking failing normalization when a source declares schema with `oneOf`?

Reply:
normalization, even prior to protocol v1, was making the assumption that the top-level schema is a plain `{type: object}`. So, yes but I don't want to fix that right now.